csrc/id_model/indexing.cpp

```diff
@@ -531,6 +528,14 @@ IndexingInfo TensorIndexer::computeIndex(
   }
 
   IndexingInfo info{loop_domains, traversal_path, index_compute.indexMap()};
+
```

Comment:
Could you give some more info why this replacement should be done here?

I'm still in the middle of refactoring, but my current plan is to have this this function as a common routine for both tensor expr indexing and predicate indexing. `getLinearIndex` is the one for expr indexing, and I'm going to have `getPredicateIndex` for predicates. I was planning to have `computeIndex` just does some common operations. The main difference of `getLinearIndex` and `getPredicateIndex` is the initial index map, so the index replacement would be done differently.